### PR TITLE
Youtube Video Player

### PR DESCRIPTION
**Pull request by mattiapicchio**

### CODE REVIEW

File: src/client/components/videoPlayer/VideoPlayer.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types';
+import React, { PureComponent } from 'react';
+import { withRouter } from 'react-router-dom';
+import YouTube from 'react-youtube';
+import styles from './videoPlayer.scss';
+
+class VideoPlayer extends PureComponent {
```
Comment (oskarleonard) [inline src/client/components/videoPlayer/VideoPlayer.js]: The logic seems very specific to youtube, maybe better to name it YoutubePlayer? Later might add a more generic VideoPlayer.

File: src/client/components/videoPlayer/VideoPlayer.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        // https://developers.google.com/youtube/player_parameters
+        autoplay: 1,
+      },
+    };
+
+    const youtubeId = this.getYoutubeIdFromUrl(episode.link);
+
+    return (
+      <div
+        className={
```
Comment (oskarleonard) [inline src/client/components/videoPlayer/VideoPlayer.js]: Lets use the `classNames` lib for better readability and flexability.
`classNames={classNames(styles.videoPlayerContainer, className)}`

File: src/client/components/videoPlayer/videoPlayer.scss [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+.videoPlayerContainer {
+  background: $color-black;
```
Comment (oskarleonard) [inline src/client/components/videoPlayer/videoPlayer.scss]: For now, i havent figured out how to handle css vars in the component lib, so lets use hardcoded colors. Or be sent in as a `style` prop. 
`< VideoPlayer style={{ backgroundColor: '#000000' }} />`

File: src/client/components/videoPlayer/videoPlayer.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  width: 100%;
+
+  .videoPlayerWrapper {
+    width: 100%;
+    max-width: 1280px;
+    margin-left: auto;
+    margin-right: auto;
+    .videoPlayer {
```
Comment (oskarleonard) [inline src/client/components/videoPlayer/videoPlayer.scss]: Line break

File: src/client/components/videoPlayer/VideoPlayer.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types';
+import React, { PureComponent } from 'react';
+import { withRouter } from 'react-router-dom';
+import YouTube from 'react-youtube';
+import styles from './videoPlayer.scss';
+
+class VideoPlayer extends PureComponent {
+  onReady = (event) => {
+    // access to player in all event handlers via event.target
```
Comment (markuslarssonlvls) [inline src/client/components/videoPlayer/VideoPlayer.js]: you need to add this callback to the props for the component as well

File: src/client/components/videoPlayer/VideoPlayer.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  getYoutubeIdFromUrl = (url = '') => {
+    const regex = new RegExp('[?&]v(=([^&#]*)|&|#|$)'),
+      results = regex.exec(url);
+    if (!results) {
+      return null;
```
Comment (markuslarssonlvls) [inline src/client/components/videoPlayer/VideoPlayer.js]: why null? Isn't it better to use undefined?

File: src/client/components/videoPlayer/VideoPlayer.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    if (!results[2]) {
+      return '';
```
Comment (markuslarssonlvls) [inline src/client/components/videoPlayer/VideoPlayer.js]: empty string? I guess you will handle this in the same way as !results

File: src/client/components/videoPlayer/VideoPlayer.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+VideoPlayer.propTypes = {
+  className: PropTypes.string,
+  episode: PropTypes.object,
```
Comment (markuslarssonlvls) [inline src/client/components/videoPlayer/VideoPlayer.js]: pass the url and not a episode

**Pull request closed without merging**